sql/sql_table.cc

```diff
@@ -8522,14 +8530,18 @@ mysql_prepare_alter_table(THD *thd, TABLE *table,
   }
   if (rename_key_list.elements)
   {
-    if ((create_info->db_type->db_type == DB_TYPE_TIANMU) &&
-        (!(thd->variables.sql_mode & MODE_NO_KEY_ERROR))) {
-      my_error(ER_TIANMU_NOT_FOUND_INDEX, MYF(0));
+    if (create_info->db_type->db_type == DB_TYPE_TIANMU){
+      sql_mode_t sql_mode = thd->variables.sql_mode;
```

Comment:
`sql_mode` assignment should be outside `if` condition.
Also, rewrite code as follows:
```
sql_mode_t sql_mode = thd->slave_thread ? global_system_variables.sql_mode : thd->variables.sql_mode;
```